xbmc/video/ContextMenus.cpp

```diff
@@ -168,4 +171,26 @@ bool CPlay::Execute(const CFileItemPtr& itemIn) const
   return true;
 };
 
+
+std::string CFavorite::GetLabel(const CFileItem& item) const
+{
+  //The 0 we're passing for contextWindow should only matter to folders
+  return g_localizeStrings.Get(XFILE::CFavouritesDirectory::IsFavourite(std::make_shared<CFileItem>(item).get(), 0) ? 14077 : 14076); // Add/Remove Favourite
```

Comment:
Intstead of 0, I would pass </code>g_WindowManager.GetActiveWindowID()</code> here.